lib/internal/inspector_async_hook.js

```diff
@@ -16,22 +16,33 @@ const hook = createHook({
     // in https://github.com/nodejs/node/pull/13870#discussion_r124515293,
     // this should be fine as long as we call asyncTaskCanceled() too.
     const recurring = true;
-    inspector.asyncTaskScheduled(type, asyncId, recurring);
+    if (type === 'PROMISE')
+      this.promises.add(asyncId);
+    else
+      inspector.asyncTaskScheduled(type, asyncId, recurring);
   },
 
   before(asyncId) {
+    if (this.promises.has(asyncId))
+      return;
     inspector.asyncTaskStarted(asyncId);
   },
 
   after(asyncId) {
+    if (this.promises.has(asyncId))
+      return;
     inspector.asyncTaskFinished(asyncId);
   },
 
   destroy(asyncId) {
+    if (this.promises.has(asyncId))
+      return this.promises.delete(asyncId);
     inspector.asyncTaskCanceled(asyncId);
   },
 });
 
+hook.promises = new Set();
```

Comment:
I know promises are properly removed from the Set when it is destroyed but from a performance standpoint would a WeakSet be more appropriate?

Reply:
@TimothyGu This stores the async ids, not the promise objects themselves … do you think I should rename this? 😄

Reply:
ohh, oops. Yeah renaming would be great, although just because I made this mistake doesn’t mean any other person would.